tv_apps/browser/js/sync/settings.js

```diff
@@ -248,28 +244,45 @@
     },
 
     onhistorychecked() {
-      var checked = this.elements.collectionHistory.checked;
+     this.oncollectionchecked(this.elements.collectionHistory,
+                              HISTORY_SETTING);
+    },
+
+    oncollectionchecked(element, setting) {
+      if (!element || !setting) {
+        return;
+      }
+
+      var checked = element.checked;
+      var settingObj = {};
+      settingObj[setting] = checked;
       if (!checked) {
-        this.collections.delete(HISTORY_SETTING);
-        navigator.mozSettings.createLock().set({
-          'sync.collections.history.enabled': checked
-        });
+        this.collections.delete(setting);
+        navigator.mozSettings.createLock().set(settingObj);
         return;
       }
 
+      // Yeah, mozId sucks...
+      // We cannot call watch twice to change the id lifetime
+      // callbacks. But we still need to do different stuff
+      // within the onlogin callback depending on who calls
+      // oncollectionchecked. So we need to save state.
+      // Thank you mozId.
+      this.setting = setting;
+      this.settingObj = settingObj;
```

Comment:
I have no idea about the ```watch``` problem, is it possible that the later ```oncollectionchecked``` call overwrites ```this.setting``` and ```this.settingObj```? Can we use an array for example ```this.settingObjs``` to queue subsequent ```oncollectionchecked``` calls and set the settings in one ```onlogin``` callback?